pkg/puller/puller_test.go

```diff
@@ -445,6 +447,44 @@ func TestDepthChange(t *testing.T) {
 	}
 }
 
+// TestContinueSyncing adds a single peer with PO 0 to hist and live sync only a peer
```

Comment:
There is still a lack of test coverage here. The `syncPeer.gone` and the `cancelBins` functionality is not being tested according to the coverage report.